### PR TITLE
docs: Add more detail and improve readability.

### PR DESCRIPTION
**Pull request by trueskawka**
Added more detail to `docs/developer-guide.md`, based on my experience with installing the app on an iOS device for the first time. Not sure if we should make it more explicit.

Improved current docs readability - shorter line lengths (mostly for `cat`ting / reading GH on a mobile device), more whitespace.

### CODE REVIEW

File: docs/developer-guide.md [hunk 1]
```diff
@@ -12,55 +14,89 @@
...
+`console.log` statements in React Native will end up in the JS console on
+this tab.
 
 ## Running on an iOS device
 1. [Set up the dev environment](#setting-up-a-dev-environment)
 2. Connect your iOS device
 3. Within the repo, `$ open ios/ZulipMobile.xcodeproj/` to open Xcode
-4. Select your device as the `build target` (from [this guide](https://facebook.github.io/react-native/docs/running-on-device-ios.html))
-5. Hit the `build and run` button
+4. Change BundleIdentifier for both ZulipMobile and ZulipMobileTests to a
```
Comment (neerajwahi) [inline docs/developer-guide.md]: This shouldn't be necessary anymore. @trueskawka do you still need to do this?

File: docs/developer-guide.md [hunk 2]
```diff
@@ -12,55 +14,89 @@
...
+6. Hit the `build and run` button (make sure your device is unlocked)
+7. If it's the first time you're running the app, you need to trust the
+developer and the app in `Settings > General > Device Management > Developer
+App` - make sure you are connected to WiFi, as it often doesn't work with
+mobile networks
+
+If it's your first time installing an app on an iOS device, you need to
+obtain Apple Developer credentials that will allow you to sign the app.
+Register at https://developer.apple.com. Then use your Apple ID in Xcode
+and choose it as your `Signing > Team` for both ZulipMobile and ZulipMobileTests.
```
Comment (neerajwahi) [inline docs/developer-guide.md]: Maybe we should add something about not committing these changes to `project.pbxproj` to diffs?

Comment (trueskawka) [inline docs/developer-guide.md]: yeah, making it more explicit will be useful :)

File: docs/developer-guide.md (hunk 1 above)
Comment (trueskawka) [inline docs/developer-guide.md]: I get this when I don't change the string, on the latest master:
signing bundle

**Pull request merged**